drafts/draft-ietf-ippm-ioam-data-integrity.xml

```diff
@@ -524,312 +500,327 @@
-      document.</t>
+        <t>The transit node updates the ICV field in the IOAM Integrity
+        Protection header. Reminder: it MUST always be a 16-octet ICV.</t>
+
+        <t>If the transit node does not add any IOAM-Data-Fields (e.g., it
+        only modifies mutable IOAM-Data-Fields or does nothing), then the
+        transit node MUST NOT update the ICV field in the IOAM Integrity
+        Protection header.</t>
+      </section>
+
+      <section anchor="IPM-Decap" title="Decapsulating node">
+        <t>In the context of integrity protection, the decapsulating node can
+        either act as a transit node (see <xref target="IPM-Transit"/> above) or
+        directly as the Validator (see <xref target="IPM-Validator"/> below).
+        The former is an alternative to put the Validator out of the forwarding
+        path in case of performance concerns.</t>
+      </section>
+
+      <section anchor="IPM-Validator" title="Validator">
+        <t>A node that performs the validation of the integrity protection is
+        referred to as the Validator. This method assumes that symmetric keys
+        have been distributed to the Validator only. The details of the
+        mechanisms used for key distribution are outside the scope of this
+        document. We refer to the method as "Zero Trust" in the sense that
+        neither the encapsulating node, nor transit nodes, nor any other
+        non-IOAM nodes need to be trusted. Indeed, the Validator is the only
+        point of trust, meaning the method is considered a full integrity
+        protection of IOAM-Data-Fields.</t>
+
+        <t>The Validator MUST recompute the ICV by iteratively following the
+        previous steps of the method in the same order, using the respective
+        symmetric keys received previously. The recomputed ICV is then compared
+        to the received ICV field. As a result, the Validator can detect if the
+        integrity of IOAM-Data-Fields is intact or altered. The validation is
+        one-step in some cases (e.g., with POT Type-0 or E2E), where only the
+        encapsulating node updates the ICV, according to the definition of this
+        method. For other cases where transit nodes also update the ICV (e.g.,
+        with Trace Option-Types), the Validator MUST identify these transit
+        nodes in order to look up their respective keys. For that, something
+        unique representing a node MUST be included in IOAM-Data-Fields (e.g.,
+        the "node_id" for Trace Option-Types). Details on how the mapping
+        between those identifiers and keys is implemented on the Validator are
+        outside the scope of this document.</t>
+
+        <t>The Validator MUST NOT update the ICV field in the IOAM Integrity
+        Protection header. Since its role is to validate the integrity of
+        IOAM-Data-Fields, it trusts itself whether it adds IOAM-Data-Fields or
+        not.</t>
```

Comment:
This is only feasible if the Validator is the decap node, correct?

Are we also dealing with cases where only a part of the overall IOAM "journey" is integrity protected?
IMHO we should exclude this in the requirements part, i.e. for IOAM integrity protection the entire header is protected and an IOAM transit node can neither add nor remove the IOAM integrity protection header,

Reply:
> This is only feasible if the Validator is the decap node, correct?

Well, yes but it could also be "another" node (that's the reason why the proposed text was kind of generic).

> Are we also dealing with cases where only a part of the overall IOAM "journey" is integrity protected?

Not sure I understand your question.

> IMHO we should exclude this in the requirements part, i.e. for IOAM integrity protection the entire header is protected

I probably missed something above, which confuses me here too. What do you mean by "the entire header is protected"? Some header fields MUST NOT be included, otherwise there is no way to recompute the ICV. Also, exclude what from (what) requirements?

> and an IOAM transit node can neither add nor remove the IOAM integrity protection header,

+1

Reply:
What I mean with "part of the journey" is the following:
Consider a network A -- B -- C --- D --- E.

Only the part B -- C -- D is across some "dangerous" territory,

Could I protect just the journey from B to C to D with the ICV?

Reply:
Got it. Indeed, we should exclude this, as a protected segment is always from the encap to the decap, not a subset.